File: backend/app/models/student.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Boolean,
    ForeignKey
)
from sqlalchemy.orm import relationship

from app.database import Base

class Student(Base):

    __tablename__ = "students"

    id = Column(Integer, primary_key=True, index=True)

    school_id = Column(Integer, ForeignKey("schools.id"), index=True)

    class_id = Column(
        Integer,
        ForeignKey("classes.id")
    )

    parent_id = Column(
        Integer,
        ForeignKey("parents.id")
    )

    first_name = Column(String)

    last_name = Column(String)

    photo = Column(String)

    face_encoding = Column(String)

    is_active = Column(Boolean, default=True)

    username = Column(String, unique=True, nullable=True, index=True)
    password_hash = Column(String, nullable=True)
    password_salt = Column(String, nullable=True)

    attendances = relationship("Attendance", back_populates="student", cascade="all, delete-orphan")
    notifications = relationship("NotificationEvent", back_populates="student", cascade="all, delete-orphan")
    school_class = relationship("Class", back_populates="students")

File: app/lib/models/student.dart
class Student {
  const Student({
    required this.id,
    required this.firstName,
    required this.lastName,
    this.classId,
    this.parentId,
    this.className,
    this.parentPhone,
    this.photo,
    this.isActive = true,
  });

  final int id;
  final int? classId;
  final int? parentId;
  final String firstName;
  final String lastName;
  final String? className;
  final String? parentPhone;
  final String? photo;
  final bool isActive;

  String get fullName => '$firstName $lastName';

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] as int,
      classId: json['class_id'] as int?,
      parentId: json['parent_id'] as int?,
      className: json['class_name'] as String?,
      parentPhone: json['parent_phone'] as String?,
      firstName: json['first_name'] as String? ?? '',
      lastName: json['last_name'] as String? ?? '',
      photo: json['photo'] as String?,
      isActive: json['is_active'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'class_id': classId,
      'parent_id': parentId,
      'first_name': firstName,
      'last_name': lastName,
      'photo': photo,
    };
  }
}
